flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobVertexFlameGraphHandler.java

```diff
@@ -63,6 +66,7 @@ public JobVertexFlameGraphHandler(
                 executionGraphCache,
                 executor);
         this.threadInfoOperatorTracker = threadInfoOperatorTracker;
+        this.isFlameGraphEnabled = isFlameGraphEnabled;
```

Comment:
if `threadInfoOperatorTracker == null` when `isFlameGraphEnabled = false`, then why do we need the extra boolean? Couldn't we just decide on `threadInfoOperatorTracker == null`?

Also: could you mark `threadInfoOperatorTracker` (field and parameter) as `@Nullable`?

Reply:
OK, I will determine whether the flame graph is enabled by checking threadInfoOperatorTracker is null or not. But it might not be better way self-interpretation. In order to minimize the confusion I added some comments.

Reply:
That may indeed also be an argument.
I wasn't 100% sure what the best way would be...I'm going to give this PR a go but will ask a colleague working on this stack to take a second look